### DISCUSSION TITLE
F4 Button No Longer Functioning

### DISCUSSION THREAD

**Discussion by soeryobadja** [simprecicchiani/ThinkPad-T460s-macOS-OpenCore#114] [Q&A] [answered]
Hi,

Thanks a lot for the great work by @duszmox and @mhl221135 for continuing this project.
I just updated to 0.6.9 from 0.6.3 and the majority of my previous problem seem to be solved. 
However, I noticed that I lost the F4 Mic Mute function here. The "Microphone Muted" icon is not showing when I press F4. Moreover, when YogaSMC App is active, then the F4 button lamp is not working (if YogaSMC App is not active, then the F4 lamp is showing) but in both case, the "Microphone Muted" icon is not showing. 
Anybody has the same problem?

Also, I still love the previous ThinkAssistant in a way when we press F9, it shows the System Preferences and not the YogaSMCPane like now. Anybody has made any modification to go back to this function? 

Thanks.

**Comment by duszmox** [comment, marked as answer]
Hi, 
The Mute Microphone button's behavior is a common bug, if you managed to get the YogaSMC pane working this issue will solve it self after a few reboots. And about the F9 button, It annoys the hell out of me too, currently I don't think there is a solution for that, I will look into it, it should be a quick fix probably if I fork the repo, but currently IO have a lot of exams, maybe in a few weeks.

**Reply by soeryobadja** [reply]
Thanks for the info. Let me look after a few more reboot. Have done a couple of reboots but the F4 still not functioning though. 

Thanks for your effort. Seems not to be a too complicated fix, but unfortunately I don't have adequate understanding in the programming and compiling details. Would love to also see if the F10 button can trigger the search function rather than Siri as well.

**Reply by soeryobadja** [reply]
It has been a few days with some reboot/shutdown, but the F4 button lamp still not working when YogaSMC App is running, but the lamp works if the YogaSMC App is not running. Moreover, in both condition (YogaSMC App running or not), no "Mic Muted" indication in macOS. Anybody with same symptom? Thanks.

**Reply by duszmox** [reply]
Now that I think about it, you probably have mismatching versions of YogaSMC kext and YogaSMCNC application, try the one before the latest version. Maybe that will solve the problem. Or update you YogaSMC kext. I will update all the kexts when the new opencore version comes out.

**Reply by soeryobadja** [reply]
Yes, this turns out to be the problem. I used different type of kext and the SMCPane. I used the correct version, and the problem solved. thanks.
